src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -106,11 +106,22 @@ private static string[] GetValidAudiences()
                 }
             }
 
-            return new string[]
+            string siteName = ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName);
+            string runtimeSiteName = ScriptSettingsManager.Instance.GetSetting(AzureWebsiteRuntimeSiteName);
+            var audiences = new List<string>
             {
-                string.Format(SiteAzureFunctionsUriFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName)),
-                string.Format(SiteUriFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName))
+                string.Format(SiteAzureFunctionsUriFormat, siteName),
+                string.Format(SiteUriFormat, siteName)
             };
+
+            if (!string.IsNullOrEmpty(runtimeSiteName) && string.Compare(siteName, runtimeSiteName, StringComparison.OrdinalIgnoreCase) != 0)
+            {
+                // on a non-production slot, the runtime site name will differ from the site name
+                // we allow both for audience
+                audiences.Add(string.Format(SiteUriFormat, runtimeSiteName));
+            }
+
+            return audiences.ToArray();
```

Comment:
Should we change return type of this function to `IEnumerable<string>`, which is what the usage needs? Then we don't need to cast to array.